rules/DeadCode/Rector/If_/RemoveDeadInstanceOfRector.php

```diff
@@ -158,10 +157,6 @@ private function isInstanceofTheSameType(Instanceof_ $instanceof): ?bool
             return null;
         }
 
-        if (! $instanceof->class instanceof FullyQualified) {
-            return null;
-        }
```

Comment:
This must stay here, as any other expression should be skipped.

Reply:
It already checked above

Reply:
I see. Please send in separate PR

Reply:
sure https://github.com/rectorphp/rector-src/pull/6096